gnomad_qc/v4/annotations/generate_variant_qc_annotations.py

```diff
@@ -0,0 +1,162 @@
+    """
+    # Initialize variant QC annotation pipeline resource collection.
+    ann_pipeline = PipelineResourceCollection(
+        pipeline_name="variant_qc_annotation",
+        overwrite=overwrite,
+    )
+
+    # Create resource collection for each step of the variant QC annotation pipeline.
+    compute_info = PipelineStepResourceCollection(
+        "--compute-info",
+        output_resources={"info_ht": get_info(split=False, test=test)},
+    )
+    split_info_ann = PipelineStepResourceCollection(
+        "--split-info",
+        output_resources={"split_info_ht": get_info(test=test)},
+        pipeline_input_steps=[compute_info],
+    )
+    export_info_vcf = PipelineStepResourceCollection(
+        "--export-info-vcf",
+        output_resources={"info_vcf": info_vcf_path(test=test)},
+        pipeline_input_steps=[compute_info],
+    )
+
+    # Add all steps to the variant QC annotation pipeline resource collection.
+    ann_pipeline.add_steps(
+        {
+            "compute_info": compute_info,
+            "split_info": split_info_ann,
+            "export_info_vcf": export_info_vcf,
+        }
+    )
+
+    return ann_pipeline
+
+
+def main(args):
+    """Generate all variant annotations needed for variant QC."""
+    hl.init(
+        default_reference="GRCh38",
+        log="/variant_qc_annotations.log",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+    test_dataset = args.test_dataset
+    test_n_partitions = args.test_n_partitions
+    test = test_dataset or test_n_partitions
+    overwrite = args.overwrite
+    resources = get_variant_qc_annotation_resources(test=test, overwrite=overwrite)
+    mt = get_gnomad_v4_vds(test=test_dataset, high_quality_only=True).variant_data
+
+    if test_n_partitions:
+        mt = mt._filter_partitions(range(test_n_partitions))
+
+    if args.compute_info:
+        # TODO: is there any reason to also compute info per platform?
+        res = resources.compute_info
+        res.check_resource_existence()
+        default_compute_info(mt, site_annotations=True).write(
+            res.info_ht.path, overwrite=overwrite
+        )
+
+    if args.split_info:
+        res = resources.split_info
+        res.check_resource_existence()
+        split_info(res.info_ht.ht()).write(res.split_info_ht.path, overwrite=overwrite)
+
+    if args.export_info_vcf:
+        res = resources.export_info_vcf
+        res.check_resource_existence()
+        hl.export_vcf(adjust_vcf_incompatible_types(res.info_ht.ht()), res.info_vcf)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--slack-channel", help="Slack channel to post results and notifications to."
+    )
+    parser.add_argument("--overwrite", help="Overwrite data", action="store_true")
+    parser.add_argument(
+        "--test-dataset", help="Use the test dataset as input.", action="store_true"
+    )
+    parser.add_argument(
+        "--test-n-partitions",
```

Comment:
Yeah, I should have thought to have you run this on frequency. I added it since I knew we might run into issues with the full list of samples, so I wanted to make sure I tested that too